erts/doc/src/erlang.xml

```diff
@@ -1618,28 +1698,58 @@ b</pre>
       <name name="error" arity="2" since=""/>
       <fsummary>Stop execution with a specified reason.</fsummary>
       <desc>
-        <p>Stops the execution of the calling process with the reason
-          <c><anno>Reason</anno></c>, where <c><anno>Reason</anno></c>
+        <p>Raises exception of class <c>error</c> with the reason
+        <c><anno>Reason</anno></c>, where <c><anno>Reason</anno></c>
           is any term. The exit reason is
           <c>{<anno>Reason</anno>, Where}</c>, where <c>Where</c>
           is a list of the functions most recently called (the current
           function first). <c><anno>Args</anno></c> is expected to be the
           list of arguments for the current function; in Beam it is used
           to provide the arguments for the current function in
           the term <c>Where</c>. As evaluating this function causes
-          the process to terminate, it has no return value.</p>
+          an exception to be raised, it has no return value.
+          The intent of the exception class <c>error</c> is to
+          signal that an unexpected error has happened (for example, a
+          function is called with a parameter that has an incorrect
+          type). See the guide about <seeguide
+          marker="system/reference_manual:errors">errors and error
+          handling</seeguide> for additional information. Example:</p>
+          <p><c>test.erl</c>:</p>
+<pre>
+-module(test).
+-export([example_fun/2]).
+
+example_fun(A1, A2) ->
+    erlang:error(my_error, [A1, A2]).
+</pre>
+<p>Erlang shell:</p>
+<pre>
+<input>6> c(test).</input>
+{ok,test}
+<input>7> test:example_fun(arg1,"this is the second argument").</input>
+** exception error: my_error
+     in function  test:example_fun/2
+         called as test:example_fun(arg1,"this is the second argument")
+ </pre>
       </desc>
     </func>
 
     <func>
```

Comment:
It can also be worth noting that if a process calls `exit(kill)` and does not catch the exception, it will terminate with exit reason `kill` and also emit exit signals with exit reason `kill` (not `killed`) to all linked processes. Such exit signals with exit reason `kill` *can* be trapped by the linked processes.

Reply:
Wow I did not know that. Then why is `kill` changed into `killed` when an `exit(P, kill)` propagates to linked processes?

Reply:
This is a design decision made a *really* long time ago and probably originates from the beginning. The only reason I can come up with is that `killed` hints about a process being "brutally" killed using `exit(P, kill)`.

The draft erlang specification http://erlang.org/download/erl_spec47.ps.gz (see section 10.6.3) describes exit signals as consisting of `Sender`, `Cause`, and a flag `ProcessCompleted`. The flag is true if the sender terminated otherwise it is false. If it is false, it implies that the signal was sent using `exit/2`. Only exit signals with `Cause=kill` and `ProcessCompleted=false` terminates the receiver regardless of its trap exit state and translate `Cause` (reason) `kill` to `killed`.

Reply:
Interesting, thanks a lot!

Reply:
I have updated the text 3b5bfbb. @rickard-green can you check my update?